frontend/src/features/stock-view/components/StockUpdateForm.tsx

```diff
@@ -0,0 +1,216 @@
+                            <Input
+                                id="quantity"
+                                label="Quantity*"
+                                type="number"
+                                placeholder="Enter quantity"
+                                value={newStock?.quantity}
+                                onChange={(event) => dispatch({ type: 'quantity', payload: event.target.value })}
+                            />
+                        </div>
+                        <Autocomplete
+                            id="uom"
+                            value={
+                                newStock && newStock.measurementUnit
+                                    ? { key: newStock.measurementUnit, value: getUnitOfMeasurement(newStock.measurementUnit) }
+                                    : null
+                            }
+                            options={UNITS_OF_MEASUREMENT}
+                            getOptionLabel={(option) => option?.value ?? ''}
+                            renderInput={(params) => <Input {...params} label="UOM*" placeholder="Select unit" />}
+                            onChange={(_, value) => dispatch({ type: 'measurementUnit', payload: value?.key ?? null })}
+                        />
+                    </div>
+                    <div className="flex items-center justify-end pt-7">
+                        <Checkbox
+                            id="isBlocked"
+                            checked={newStock?.isBlocked ?? false}
+                            onChange={(event) => dispatch({ type: 'isBlocked', payload: event.target.checked })}
+                        />
+                        <label htmlFor="isBlocked"> is Blocked </label>
+                    </div>
+                </div>
+                <div className="w-[32rem]">
+                    <Autocomplete
+                        id="site"
+                        value={newStock?.stockLocationBpns ?? null}
+                        options={sites?.map((site) => site.bpns) ?? []}
+                        renderInput={(params) => <Input {...params} label="Stock Location BPNS*" placeholder="Select a Site" />}
+                        onChange={(_, value) => dispatch({ type: 'stockLocationBpns', payload: value ?? null })}
+                    />
+                    <Autocomplete
+                        id="site"
+                        value={newStock?.stockLocationBpna ?? null}
+                        options={sites?.find((site) => site.bpns === newStock?.stockLocationBpns)?.addresses.map((addr) => addr.bpna) ?? []}
+                        renderInput={(params) => <Input {...params} label="Stock Location BPNA*" placeholder="Select an Address" />}
+                        onChange={(_, value) => dispatch({ type: 'stockLocationBpna', payload: value ?? null })}
+                    />
+                    <div className="grid grid-cols-2 gap-2">
+                        <Input
```

Comment:
Dynamics are missing: I'm able to input a customer order position number without customer order number (when setting both, I can remove the customer order number but position number remains)

Dynamics come from samm mandatoryness: 

customer order position reference (customer order number, customer order position, number, supplier order number) are optional. if reference is given, customer order number and customer order position number are mandatory